drivers/ieee802154/ieee802154_gecko.c

```diff
@@ -0,0 +1,588 @@
+
+	switch (mode) {
+	case IEEE802154_TX_MODE_DIRECT:
+		status = RAIL_StartTx(dev_data->rail_handle, dev_data->channel,
+				RAIL_TX_OPTIONS_DEFAULT, NULL);
+		break;
+	case IEEE802154_TX_MODE_CCA:
+		status = RAIL_StartCcaLbtTx(dev_data->rail_handle,
+				dev_data->channel,
+				RAIL_TX_OPTIONS_DEFAULT,
+				&rail_lbt_config,
+				NULL);
+		break;
+	case IEEE802154_TX_MODE_CSMA_CA:
+		status = RAIL_StartCcaCsmaTx(dev_data->rail_handle,
+				dev_data->channel,
+				RAIL_TX_OPTIONS_DEFAULT,
+				&rail_csma_config, NULL);
+		break;
+	case IEEE802154_TX_MODE_TXTIME:
+	case IEEE802154_TX_MODE_TXTIME_CCA:
+	default:
+		NET_ERR("TX mode %d not supported", mode);
+		return -ENOTSUP;
+	}
+
+	if (status != RAIL_STATUS_NO_ERROR) {
+		LOG_ERR("Failed to start Tx");
+		return -EIO;
+	}
+
+	/* Wait for the callback from the radio driver. */
+	if (0 != k_sem_take(&dev_data->tx_wait, TX_PACKET_SENT_TIMEOUT)) {
+		LOG_DBG("Failed to take tx_wait semaphore");
+		return -EIO;
+	}
+
+	return dev_data->tx_status;
+}
+
+#define RAIL_IRQ_PRIO  0
+
+void ieee802154_gecko_irq_config(void)
+{
+	IRQ_DIRECT_CONNECT(RFSENSE_IRQn, RAIL_IRQ_PRIO, RFSENSE_IRQHandler, 0);
+	irq_enable(RFSENSE_IRQn);
+	IRQ_DIRECT_CONNECT(AGC_IRQn, RAIL_IRQ_PRIO, AGC_IRQHandler, 0);
+	irq_enable(AGC_IRQn);
+	IRQ_DIRECT_CONNECT(BUFC_IRQn, RAIL_IRQ_PRIO, BUFC_IRQHandler, 0);
+	irq_enable(BUFC_IRQn);
+	IRQ_DIRECT_CONNECT(FRC_IRQn, RAIL_IRQ_PRIO, FRC_IRQHandler, 0);
+	irq_enable(FRC_IRQn);
+	IRQ_DIRECT_CONNECT(FRC_IRQn, RAIL_IRQ_PRIO, FRC_IRQHandler, 0);
+	irq_enable(FRC_IRQn);
+	IRQ_DIRECT_CONNECT(FRC_PRI_IRQn, RAIL_IRQ_PRIO, FRC_PRI_IRQHandler, 0);
+	irq_enable(FRC_PRI_IRQn);
+	IRQ_DIRECT_CONNECT(MODEM_IRQn, RAIL_IRQ_PRIO, MODEM_IRQHandler, 0);
+	irq_enable(MODEM_IRQn);
+	IRQ_DIRECT_CONNECT(PROTIMER_IRQn, RAIL_IRQ_PRIO, PROTIMER_IRQHandler, 0);
+	irq_enable(PROTIMER_IRQn);
+	IRQ_DIRECT_CONNECT(RAC_RSM_IRQn, RAIL_IRQ_PRIO, RAC_RSM_IRQHandler, 0);
+	irq_enable(RAC_RSM_IRQn);
+	IRQ_DIRECT_CONNECT(RAC_SEQ_IRQn, RAIL_IRQ_PRIO, RAC_SEQ_IRQHandler, 0);
+	irq_enable(RAC_SEQ_IRQn);
+	IRQ_DIRECT_CONNECT(SYNTH_IRQn, RAIL_IRQ_PRIO, SYNTH_IRQHandler, 0);
+	irq_enable(SYNTH_IRQn);
+}
+
+static int ieee802154_gecko_init_rail(struct device *dev)
+{
+	struct ieee802154_gecko_context *dev_data = DEV_DATA(dev);
+	const struct ieee802154_gecko_dev_cfg *const dev_cfg = DEV_CFG(dev);
+	RAIL_Status_t status;
+
+	dev_data->rail_handle = RAIL_Init(&rail_config, NULL);
+	if (dev_data->rail_handle == NULL) {
+		LOG_DBG("Failed to get RAIL handle");
+		return -EIO;
+	}
+	RAIL_Idle(dev_data->rail_handle, RAIL_IDLE, true);
+
+	RAIL_EnablePaCal(true);
+	RAIL_ConfigCal(dev_data->rail_handle, RAIL_CAL_ALL);
+
+	/* Configure RAIL callbacks */
+	RAIL_ConfigEvents(dev_data->rail_handle, RAIL_EVENTS_ALL,
+			  RAIL_EVENT_RX_PACKET_RECEIVED |
+			  RAIL_EVENT_TX_PACKET_SENT |
+			  RAIL_EVENT_TX_ABORTED |
+			  RAIL_EVENT_TX_BLOCKED |
+			  RAIL_EVENT_TX_UNDERFLOW |
+			  RAIL_EVENT_TX_CHANNEL_BUSY |
+			  RAIL_EVENT_CAL_NEEDED);
+
```

Comment:
```suggestion
const RAIL_TxPowerCurvesConfigAlt_t tx_power_curves = RAIL_DECLARE_TX_POWER_CURVES_CONFIG_ALT;
RAIL_InitTxPowerCurvesAlt(&tx_power_curves);
```

I think it is used to properly calculate the TX power.